Simplify deployment in federation CI jobs

shashidharatd opened the pull request:
**What this PR does / why we need it**:
Introduce a new variable `USE_PV_FOR_ETCD` to control whether to use PV for federation etcd. `USE_PV_FOR_ETCD` is set to false by default and can be overridden in CI jobs where we want to test with PV. For most of the federation CI jobs, it does not matter whether we use PV for etcd or not. We can overcome some issues by not depending on PV's in federation setup like failure to bind to PV and leaking PD's. etc...
#50543 is one such issue associated with PV.

This PR also contain a commit to fix couple of minor issues and removing unnecessary include of cluster/common script.

**Release note**:
```release-note
NONE
```

/assign @madhusudancs 
/cc @kubernetes/sig-federation-pr-reviews

Commit:
Add switch to control use of pv for etcd in federation CI jobs

Commit:
Fix couple of minor issues in federation deployment scripts

Commit:
Remove unnecessarily included scripts in federation deployment scripts

shashidharatd commented:
/test pull-kubernetes-unit

shashidharatd commented:
@madhusudancs, PTAL at this PR.

madhusudancs commented on federation/cluster/federation-up.sh:
```diff
@@ -118,6 +115,7 @@ function init() {
       --apiserver-enable-token-auth=true \
       --apiserver-arg-overrides="--runtime-config=api/all=true,--v=4" \
       --controllermanager-arg-overrides="--v=4" \
+      --etcd-persistent-storage=${USE_PV_FOR_ETCD:-false} \
```
Can you please do this defaulting at the top? See https://github.com/kubernetes/kubernetes/blob/master/federation/cluster/common.sh#L25 for example. It's difficult to track it otherwise. Also, can you please prefix the global var passed down from the test infra with `FEDERATION_` to not pollute the namespace.

madhusudancs reviewed (commented):
@shashidharatd Minor nit. Let me know once you address that. I will LGTM the PR.

/approve

shashidharatd commented:
/test pull-kubernetes-federation-e2e-gce
/test pull-kubernetes-e2e-kops-aws

Commit:
Move logs related to etcd pvc creation inside conditional

shashidharatd commented:
/test pull-kubernetes-federation-e2e-gce

shashidharatd commented:
@madhusudancs, i have addressed your comment and `pull-kubernetes-federation-e2e-gce` pre-submit job is green. `pull-kubernetes-e2e-kops-aws` is failing for some test-infra issue. 
Can you PTAL. Thanks

madhusudancs reviewed (commented):
/lgtm

madhusudancs commented:
/retest
